lab_5_scrapper/scrapper.py

```diff
@@ -13,70 +57,138 @@ def __init__(self, path_to_config: Path) -> None:
+        headers = info['headers']
+        total_articles_to_find_and_parse = info['total_articles_to_find_and_parse']
+        encoding = info['encoding']
+        timeout = info['timeout']
+        verify_certificate = info['should_verify_certificate']
+        headless_mode = info['headless_mode']
+
+        if not isinstance(seed_urls, list):
+            raise IncorrectSeedURLError
+
+        for url in seed_urls:
+            result = urlparse(url)
+            if (not isinstance(url, str)
+                    or not result.netloc or not result.scheme):
+                raise IncorrectSeedURLError
+
+        if (not isinstance(total_articles_to_find_and_parse, int)
+                or isinstance(total_articles_to_find_and_parse, bool)
+                or total_articles_to_find_and_parse < 1):
+            raise IncorrectNumberOfArticlesError
+
+        if total_articles_to_find_and_parse > NUM_ARTICLES_UPPER_LIMIT:
+            raise NumberOfArticlesOutOfRangeError
+
+        if not isinstance(headers, dict):
+            raise IncorrectHeadersError
+
+        if not isinstance(encoding, str):
+            raise IncorrectEncodingError
+
+        if (not isinstance(timeout, int)
+                or timeout < TIMEOUT_LOWER_LIMIT
+                or timeout > TIMEOUT_UPPER_LIMIT):
+            raise IncorrectTimeoutError
+
+        if not isinstance(verify_certificate, bool):
+            raise IncorrectVerifyError
+
+        if not isinstance(headless_mode, bool):
+            raise IncorrectVerifyError
 
     def get_seed_urls(self) -> list[str]:
         """
         Retrieve seed urls
         """
-        pass
+        return self._seed_urls
 
     def get_num_articles(self) -> int:
         """
         Retrieve total number of articles to scrape
         """
-        pass
+        return self._num_articles
 
     def get_headers(self) -> dict[str, str]:
         """
         Retrieve headers to use during requesting
         """
-        pass
+        return self._headers
 
     def get_encoding(self) -> str:
         """
         Retrieve encoding to use during parsing
         """
-        pass
+        return self._encoding
 
     def get_timeout(self) -> int:
         """
         Retrieve number of seconds to wait for response
         """
-        pass
+        return self._timeout
 
     def get_verify_certificate(self) -> bool:
         """
         Retrieve whether to verify certificate
         """
-        pass
+        return self._should_verify_certificate
 
     def get_headless_mode(self) -> bool:
         """
         Retrieve whether to use headless mode
         """
-        pass
+        return self._headless_mode
 
 
 def make_request(url: str, config: Config) -> requests.models.Response:
     """
     Delivers a response from a request
     with given configuration
     """
-    pass
+    determined_pause = 0.5
+    divider = 2
+    time.sleep(determined_pause + random.random() / divider)
```

Comment:
can you please describe why you use this time logic with division? is simple time call not helpful?

Reply:
fixed